distribution/lib/Standard/Database/0.0.0-dev/src/Internal/JDBC_Connection.enso

```diff
@@ -287,3 +301,18 @@ handle_sql_errors ~action related_query=Nothing =
 set_statement_values stmt statement_setter values =
     values.each_with_index ix-> value->
         statement_setter.fill_hole stmt (ix + 1) value
+
+## PRIVATE
+   A helper that logs performed SQL queries/statements to a file, if an
+   environment variable is set.
+log_sql_if_enabled : JDBC_Connection -> Text -> Nothing
+log_sql_if_enabled jdbc_connection ~query_text =
+    case Environment.get "ENSO_SQL_LOG_PATH" of
+        Nothing -> Nothing
+        path -> if path.is_empty then Nothing else
+            timestamp = Date_Time.now.format "yyyy-MM-dd HH:mm:ss zzzzz"
+            db_id = jdbc_connection.with_metadata .getDatabaseProductName
+            # E.g. 2023-10-09 11:27:08 +02:00 [SQLite] SELECT * FROM "test"
+            log_line = timestamp + " [" + db_id + "] " + query_text + '\n'
+            Context.Output.with_enabled <|
+                log_line.write path on_existing_file=Existing_File_Behavior.Append
```

Comment:
I was considering using `getUrl` instead of `getDatabaseProductName` to more specifically identify which connection these operations are for.

But I was afraid that the URL could expose secrets like a password (normally we set it thru properties, but _it is possible_ to set it thru URL and so it could leak). So to avoid that I went for a middle ground and I'm just logging the database name.

Does that sound OK? Or do we prefer a full URL? It is mostly for internal debug use anyway, so users should know what they are doing when setting this env var. Still, setting it already exposes some data to anyone who has control over the environment variables.

Reply:
Using the Database platform is perfect.

Reply:
We disable sql logging in **our** logging infrastructure (https://github.com/enso-org/enso/blob/develop/engine/language-server/src/main/resources/application.conf#L23) because that would create a lot of (I really mean a lot) of unnecessary output. You could control that via custom env vars as described in https://github.com/enso-org/enso/blob/develop/docs/infrastructure/logging.md#custom-log-levels